analyzers/src/SonarAnalyzer.CSharp/Metrics/CSharpCognitiveComplexityMetric.cs

```diff
@@ -27,6 +27,10 @@ public static class CSharpCognitiveComplexityMetric
 
         public static CognitiveComplexity GetComplexity(SyntaxNode node, bool onlyGlobalStatements)
         {
+            if (GeneratedCodeRecognizer.IsRazor(node.SyntaxTree))
+            {
+                return new(0, ImmutableArray<SecondaryLocation>.Empty);
+            }
```

Comment:
This will disable also the cognitive complexity rule, not only the metric. This rule is not exclude by: https://github.com/SonarSource/sonar-dotnet/issues/7879. Could you please check with @csaba-sagi-sonarsource if it makes sense to keep the rule active or not for razor files?

Reply:
Personal opinion: If we don't have a reliable count of cognitive complexity for .razor files to be shown in the metrics why would we want a rule that might be raised based on the same flawed logic?

Reply:
The rule is computing the complexity at the function level. For this kind of limited scope, the precision is better. Please check with @csaba-sagi-sonarsource as he has investigated this already.

Reply:
I've checked with @csaba-sagi-sonarsource and he thinks that the count for the complexity of methods fully declared inside .razor files should be good enough. I changed the implementation to target the metrics only.